src/irr/asset/CImageWriterJPG.cpp

```diff
@@ -102,6 +102,8 @@ static void jpeg_file_dest(j_compress_ptr cinfo, io::IWriteFile* file)
 static bool writeJPEGFile(io::IWriteFile* file, const asset::CImageData* image, uint32_t quality)
 {
 	void (*format)(const void*, int32_t, void*) = 0;
+	bool grayscale = false;
+	
 	switch( image->getColorFormat () )
 	{
         case asset::EF_R8G8B8_UNORM:
```

Comment:
none of the below code with CColorConverter will work, that conversion code is legacy